web/src/WifiSelector.jsx

```diff
@@ -0,0 +1,276 @@
+                  label={n.ssid}
+                  description={
+                    <>
+                      <LockIcon size="10" color="grey" /> {n.security.join(", ")}{" "}
+                      <SignalIcon size="10" color="grey" /> {n.strength}
+                    </>
+                  }
+                  isChecked={isChecked || showAsChecked || false}
+                  onClick={() => {
+                    if (isChecked) return;
+                    switchSelectedNetwork(n);
+                    if (n.settings && !n.connection) client.network.connectTo(n.settings);
+                  }}
+                />
+              </SplitItem>
+              <SplitItem>
+                <Center>
+                  {showSpinner && <Spinner isSVG size="md" aria-label={`${n.ssid} connection is waiting for an state change`} /> }
+                </Center>
+              </SplitItem>
+              <SplitItem>
+                <Center>
+                  <Text component="small" className="keep-words">
+                    { showSpinner && !n.connection && "Connecting" }
+                    { networkState(n.connection?.state)}
+                  </Text>
+                </Center>
+              </SplitItem>
+              { n.settings &&
+                <SplitItem>
+                  <Center>
+                    <WifiNetworkMenu settings={n.settings} />
+                  </Center>
+                </SplitItem> }
+            </Split>
+            { isChecked && (!n.settings || n.settings.error) &&
+              <Split hasGutter>
+                <SplitItem isFilled className="content">
+                  <WifiConnectionForm network={n} onCancel={() => switchSelectedNetwork(activeNetwork)} />
+                </SplitItem>
+              </Split> }
+          </CardBody>
+        </Card>
+      );
+    });
+  };
+
+  const renderHiddenNetworkForm = () => {
+    return (
+      <>
+        <Card className={showHiddenForm ? "selection-list-item selection-list-focused-item" : "selection-list-item collapsed"}>
+          <CardBody>
+            <Split hasGutter className="content">
+              <SplitItem isFilled>
+                { showHiddenForm &&
+                  <WifiConnectionForm
+                    network={selectedNetwork}
+                    onCancel={() => switchSelectedNetwork(activeNetwork)}
+                    onSubmitCallback={switchSelectedNetwork}
+                  /> }
+              </SplitItem>
+            </Split>
+          </CardBody>
+        </Card>
+        { !showHiddenForm &&
+          <Center>
+            <Button
+              variant="link"
+              onClick={() => switchSelectedNetwork(baseHiddenNetwork)}
+            >
+              Connect to hidden network
+            </Button>
+          </Center> }
+      </>
+    );
+  };
+
+  return (
+    <Popup isOpen={isOpen} height="large" title="Connect to a Wi-Fi network">
+      { renderFilteredNetworks(networksFromValues(networks)) }
```

Comment:
Why not use a proper component? OK, I know, components are functions, but I would expect to have something like `Networks` and `HiddenNetworkForm`. Do you need to share all that state between them?

In any case, about the `renderFilteredNetworks`, does that function do any filtering? If that's not the case, I do not think you need to mention that in the name. It isn't very clear.

Reply:
Imo, you're right.

Somehow, we forget to extract this to small components after so many changes to get the Wi-Fi selector working. I have [tried to improve the situation a bit](https://github.com/yast/d-installer/pull/292/commits/beb0e4a7ffcf41c5d41e1c4facb9c546e2028472), but I fear a better improvement/re-organization must wait until next iterations in the network area.

What is more, I'd like to create a custom _SelectionList_ component (Wi-Fi and Product selector are quite similar, though) and reorganize not only that part, but the whole src/ directory. 

Hope, however, it looks good/acceptable enough for now.